src/Workspaces/SharedUtilitiesAndExtensions/Workspace/CSharp/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2629,8 +2629,16 @@ public static bool IsNameOfContext(this SyntaxTree syntaxTree, int position, Sem
             }
 
             // is/as are valid after expressions.
-            if (token.IsLastTokenOfNode<ExpressionSyntax>())
+            if (token.IsLastTokenOfNode<ExpressionSyntax>(out var expression))
             {
+                // 'is/as' not allowed after a anonymous-method/lambda/method-group.
+                if (expression.IsAnyLambdaOrAnonymousMethod())
+                    return false;
+
+                var symbol = semanticModel.GetSymbolInfo(expression).GetAnySymbol();
+                if (symbol is IMethodSymbol)
+                    return false;
```

Comment:
❔ Is this a commonly-hit case? What is the likelihood that this symbol check will cause a performance problem on the typing path?

Reply:
practically none.  We do tons more semantic ops all of the place when computing keyword intellisense.  we'll basically have already bound the nodes around here and this will be practically free.